### PR TITLE
add sentinel file to detect whether bundled or not

### LINKED ISSUE #3596: running_as_bundled_app no longer works
## 🐛 Bug
The `running_as_bundled_app`  function no longer seems to work, and the reason is that the METADATA file currently produced (at `app/napari-...dist-info/METADATA`) is extremely minimal, and lacks the stuff (like Briefcase-Version) that is "supposed" to be in there as per https://briefcase.readthedocs.io/en/latest/how-to/access-packaging-metadata.html?highlight=briefcase-version.

@goanpeca, @jaimergp ... do you have any ideas why that might be?
Perhaps we can come up with another strategy?  like adding a package metadata flag ourselves (in `bundle.py` that we could look for?

### PR DESCRIPTION
**Pull request by jaimergp**
# Description

## Type of change

- [X] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

Closes #3596 

# How has this been tested?

- [ ] It was working on #3555 :)

## Final checklist:
- [X] My PR is the minimum possible work for the desired functionality
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
Lgtm! Thanks

File: bundle.py [hunk 1]
```diff
@@ -205,6 +206,17 @@ def patch_python_lib_location():
         print("symlinking", orig, "to", dest)
 
 
+def add_sentinel_file():
+    if MACOS:
+        (Path(APP_DIR) / "Contents" / "MacOS" / ".napari_is_bundled").touch()
```
Comment (tlambert03) [inline bundle.py:213]: looks like `APP_DIR` is only defined on MACOS?

**Comment by sofroniewn** [comment]
The bundle creation test looks like it is failing

**Comment by jaimergp** [comment]
Ooops, I missed that part of the changes. Let's see now :)

**Review by goanpeca — APPROVED** [review]
🥳

**Pull request merged**